lib/connection.go

```diff
@@ -162,6 +162,7 @@ func (c *Conn) NewRequest(method, path, query string) (*Request, error) {
 		uri = fmt.Sprintf("%s://%s:%s%s", c.Protocol, host, portNum, path)
 	}
 	req, err := http.NewRequest(method, uri, nil)
+	hr.Mark(err)
```

Comment:
I think this is the wrong spot, NewRequest doesn't actually perform the request, it just creates the object. It'll only error if the method is invalid or it fails to parse the URI or whatever

Reply:
Humm your right.  Need to move this up a level

Reply:
I think you want it here https://github.com/launchdarkly/elastigo/blob/master/lib/baserequest.go#L78
I think you should be able to do req.hostResponse.Mark since it's in the same package